geomesa-filter/src/main/scala/org/locationtech/geomesa/filter/expression/FastPropertyIsEqualTo.scala

```diff
@@ -23,6 +23,18 @@ abstract class FastPropertyIsEqualTo(exp1: Expression, exp2: Literal) extends Pr
   override def getMatchAction: MatchAction = MatchAction.ANY
 
   override def toString: String = s"[ $exp1 = $exp2 ]"
+
+  def canEqual(other: Any): Boolean = other.isInstanceOf[FastPropertyIsEqualTo]
+
+  override def equals(other: Any): Boolean = other match {
```

Comment:
Errm... why do we have to implement equals and hashCode?

Is it because we've got our own type?

Reply:
makes unit tests pass... may be other benefits i guess